core/src/main/java/org/operationcrypto/hivej/communication/HttpClient.java

```diff
@@ -17,59 +17,105 @@
 package org.operationcrypto.hivej.communication;
 
 import java.io.IOException;
-import java.net.URI;
+import java.net.URL;
 import java.security.GeneralSecurityException;
-
-import org.apache.http.client.ClientProtocolException;
-import org.operationcrypto.hivej.jrpc.JsonRPCRequest;
-import org.operationcrypto.hivej.jrpc.JsonRPCResponse;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import java.util.UUID;
 
 import com.google.api.client.http.ByteArrayContent;
 import com.google.api.client.http.GenericUrl;
 import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestFactory;
 import com.google.api.client.http.HttpResponse;
 import com.google.api.client.http.javanet.NetHttpTransport;
 
+import org.apache.http.client.ClientProtocolException;
+import org.operationcrypto.hivej.jrpc.JsonRPCRequest;
+import org.operationcrypto.hivej.jrpc.JsonRPCResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 /**
  * This class handles the communication to a Hive Node using the HTTP protocol.
  * 
  * @author <a href="https://github.com/marvin-we">marvin-we</a>
  */
-public class HttpClient implements AbstractClient {
+public class HttpClient extends AbstractClient {
     private static final Logger LOGGER = LoggerFactory.getLogger(HttpClient.class);
+    private final String CLIENT_ID;
```

Comment:
why is CLIENT_ID uppercase? shouldnt it match the usual regex pattern?

Reply:
'final' variables should be uppercase, no?

Reply:
Even Sonar complains about that one - with the rule you just stated above (https://rules.sonarsource.com/java/tag/convention/RSPEC-116). Writing final variables uppercase is fine to me, but could introduce a rule for that to make sonar happy?